pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -482,10 +486,14 @@ func (gc *GarbageCollector) Run(workers int, stopCh <-chan struct{}) {
 	gc.propagator.eventQueue.ShutDown()
 }
 
+// QueueDrained returns if the dirtyQueue and eventQueue are drained. It's
+// useful for debugging.
 func (gc *GarbageCollector) QueuesDrained() bool {
 	return gc.dirtyQueue.Len() == 0 && gc.propagator.eventQueue.Len() == 0
 }
 
+// GraphHasUID returns if the Propagator has a particular UID store in its
+// uidToNode graph. It's useful for debugging.
 func (gc *GarbageCollector) GraphHasUID(UIDs []types.UID) bool {
```

Comment:
might this need a lock? maybe the comment should say tests only, or that it shouldn't be called if the garbage collector is busy?